lib/internal/watch_mode/files_watcher.js

```diff
@@ -74,17 +75,19 @@ class FilesWatcher extends EventEmitter {
   }
 
   #onChange(trigger) {
-    if (this.#debouncing.has(trigger)) {
-      return;
-    }
     if (this.#mode === 'filter' && !this.#filteredFiles.has(trigger)) {
       return;
     }
-    this.#debouncing.add(trigger);
+
     const owners = this.#depencencyOwners.get(trigger);
-    setTimeout(() => {
-      this.#debouncing.delete(trigger);
+    if (owners) for (const o of owners) this.#debounceOwners.add(o);
```

Comment:
Can you make this a little bit readable?

```suggestion
    if (owners) {
      for (const owner of owners) {
        this.#debounceOwners.add(owner);
      }
    }
```